skimage/filters/rank/generic.py

```diff
@@ -164,18 +270,18 @@ def func_out(image, selem=None, *args, **kwargs):
 def autolevel(image, selem, out=None, mask=None, shift_x=False, shift_y=False):
     """Auto-level image using local histogram.
 
-    This filter locally stretches the histogram of greyvalues to cover the
+    This filter locally stretches the histogram of grey values to cover the
     entire range of values from "white" to "black".
 
     Parameters
     ----------
-    image : 2-D array (uint8, uint16)
+    image : 2-D array (integer, float or boolean)
         Input image.
-    selem : 2-D array
+    selem : 2-D array (integer, float or boolean)
```

Comment:
Please, add the missing ", optional" in this docstring.

Reply:
Can you do it also in the other docstrings below?

Reply:
I'm not sure which fields should be marked optional. Are you referring to all the `selem` fields or all the fields below?

Reply:
the ones that typically have `=None` where in the function call, they get assigned default values.

Reply:
By that logic, should the `selem` arg have a default `=None` in the function call as well?

Reply:
in some of them it seems so.